integration-tests/src/tests/reserve_backed_transfers.rs

```diff
@@ -194,71 +203,55 @@ fn reserve_to_polimec() {
 	let max_weight = Weight::from_parts(MAX_REF_TIME, MAX_PROOF_SIZE);
 
-	// Give some usdt to Polimec on Statemint
-	mint_usdt_on_statemint_to(&polimec_sibling_account, RESERVE_TRANSFER_AMOUNT * 3);
-	// Represent that usdt just minted in Polimec, by minting to Alice
-	Polimec::execute_with(|| {
-		assert_ok!(PolimecStatemintAssets::mint_into(
-			AcceptedFundingAsset::USDT.to_statemint_id(),
-			&alice_account,
-			RESERVE_TRANSFER_AMOUNT * 3
-		));
-		assert_ok!(PolimecStatemintAssets::mint_into(
-			AcceptedFundingAsset::DOT.to_statemint_id(),
-			&alice_account,
-			RESERVE_TRANSFER_AMOUNT * 3
-		));
+	mint_asset_on_asset_hub_to(asset_id, &polimec_sibling_account, RESERVE_TRANSFER_AMOUNT + 1_0_000_000_000);
+
+	PolimecBase::execute_with(|| {
+		assert_ok!(BaseForeignAssets::mint_into(asset_id, &alice_account, RESERVE_TRANSFER_AMOUNT + 1_0_000_000_000));
 	});
 
-	// Check Polimec's pre transfer balances and issuance
 	let (
-		polimec_prev_alice_usdt_balance,
+		polimec_prev_alice_asset_balance,
 		polimec_prev_alice_plmc_balance,
-		polimec_prev_usdt_issuance,
+		polimec_prev_asset_issuance,
 		polimec_prev_plmc_issuance,
-	) = Polimec::execute_with(|| {
-		(
-			PolimecStatemintAssets::balance(AcceptedFundingAsset::USDT.to_statemint_id(), alice_account.clone()),
-			PolimecBalances::free_balance(alice_account.clone()),
-			PolimecStatemintAssets::total_issuance(AcceptedFundingAsset::USDT.to_statemint_id()),
-			PolimecBalances::total_issuance(),
-		)
-	});
-
-	// check Statemint's pre transfer balances and issuance
-	let (statemint_prev_alice_usdt_balance, statemint_prev_usdt_issuance) = Statemint::execute_with(|| {
-		(
-			StatemintAssets::balance(AcceptedFundingAsset::USDT.to_statemint_id(), alice_account.clone()),
-			StatemintAssets::total_issuance(AcceptedFundingAsset::USDT.to_statemint_id()),
-		)
-	});
-
-	let usdt_plus_exec_fee: MultiAsset = (usdt_on_statemint, RESERVE_TRANSFER_AMOUNT + 1_0_000_000_000).into();
-	let dot_for_xcm_execution: MultiAsset = (Polimec::parent_location(), 1_0_000_000_000u128).into();
-	// construct the XCM to transfer from Polimec to Statemint's reserve
-	let transfer_xcm: Xcm<PolimecCall> = Xcm(vec![
-		WithdrawAsset(vec![usdt_plus_exec_fee.clone(), dot_for_xcm_execution.clone()].into()),
-		BuyExecution { fees: usdt_plus_exec_fee.clone(), weight_limit: Limited(max_weight) },
+	) = get_polimec_balances(asset_id, alice_account.clone());
+
+	// check AssetHub's pre transfer balances and issuance
+	let (asset_hub_prev_alice_asset_balance, asset_hub_prev_polimec_asset_balance, asset_hub_prev_asset_issuance) =
+		get_asset_hub_balances(asset_id, alice_account.clone(), polimec_sibling_account.clone());
+
+	let transferable_asset_plus_exec_fee: MultiAsset =
+		(asset_hub_asset_id, RESERVE_TRANSFER_AMOUNT + 1_0_000_000_000).into();
+	let mut asset_hub_exec_fee: MultiAsset = (asset_hub_asset_id.clone(), 1_0_000_000_000u128).into();
```

Comment:
Can you pay for execution in the AH with USDC and USDT?

Reply:
Yes they have a trader that can use the first token in the holding as fee payment token, if the token exists in their assets pallet. So USDC and USDT can be used to trade.